### ISSUE TITLE
Campfire world editing causing chunk errors

### ISSUE THREAD

**Issue by KourisBP** [IntellectualSites/FastAsyncWorldEdit#415]
# Bug report for FastAsyncWorldEdit 1.15.2

**[REQUIRED] FastAsyncWorldEdit Configuration Files:** 
https://pastebin.com/9PDsXwbx

**[REQUIRED] FastAsyncWorldEdit Version Number:** 
Build 172

**[REQUIRED] Spigot/Paper Version Number:** 1.15.2

**[REQUIRED] Description of the problem:**

World editing Campfires creates chunk errors in order to fix it I had to delete the entire chunk
**How to replicate**:

I world edited a campfire that was part of a house with copy and paste restarted the server and the world wouldn't load as it had chunk errors

**Plugins being used on the server:**

https://prnt.sc/s2x9yc
**Checklist**:

- [X] I included all information required in the sections above
- [X] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit/issues?q=is%3Aissue)
- [X] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.15.2](https://ci.athion.net/job/FastAsyncWorldEdit-1.15/)
- [X] I made sure the bug/error is not caused by any other plugin

**Comment by dordsor21** [comment]
The error log would be useful...

**Comment by KourisBP** [comment]
[latest.log](https://github.com/IntellectualSites/FastAsyncWorldEdit/files/4509774/latest.log)
It seems like its not just campfires but random entities being worldedited seem to cause chunk errors but this one seemed to somehow fix itself but it only happens to blocks that are worldedit and that I'm sure of

**Comment by Sauilitired** [comment]
Relevant excerpt:

```
[13:33:14] [Server thread/WARN]: com.destroystokyo.paper.exception.ServerInternalException: Attempted to place a tile entity (net.minecraft.server.v1_15_R1.TileEntityLectern@1bb8a092) at -22,75,358 (Block{minecraft:air}) where there was no entity tile!
[13:33:14] [Server thread/WARN]: Chunk coordinates: -32,352
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.setTileEntity(Chunk.java:593)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.a(Chunk.java:565)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkRegionLoader.loadEntities(ChunkRegionLoader.java:578)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkRegionLoader.lambda$loadChunk$6(ChunkRegionLoader.java:161)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.addEntities(Chunk.java:648)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.PlayerChunkMap.lambda$null$28(PlayerChunkMap.java:801)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either.mapLeft(Either.java:162)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.PlayerChunkMap.lambda$c$29(PlayerChunkMap.java:784)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkTaskQueueSorter.b(SourceFile:58)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer$a.executeTask(ChunkProviderServer.java:893)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer$a.executeNext(ChunkProviderServer.java:905)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:396)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:325)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IWorldReader.getChunkAt(IWorldReader.java:87)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:270)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.loadSpawn(MinecraftServer.java:618)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.createWorld(CraftServer.java:1045)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.Bukkit.createWorld(Bukkit.java:536)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.WorldCreator.createWorld(WorldCreator.java:320)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:467)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:448)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:423)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.loadWorlds(WorldManager.java:728)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.MultiverseCore.onEnable(MultiverseCore.java:310)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:470)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:384)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:482)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:298)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885)
[13:33:14] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[13:33:14] [Server thread/WARN]: com.destroystokyo.paper.exception.ServerInternalException: Attempted to place a tile entity (net.minecraft.server.v1_15_R1.TileEntityLectern@c7dc40d) at -23,75,358 (Block{minecraft:air}) where there was no entity tile!
[13:33:14] [Server thread/WARN]: Chunk coordinates: -32,352
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.setTileEntity(Chunk.java:593)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.a(Chunk.java:565)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkRegionLoader.loadEntities(ChunkRegionLoader.java:578)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkRegionLoader.lambda$loadChunk$6(ChunkRegionLoader.java:161)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.Chunk.addEntities(Chunk.java:648)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.PlayerChunkMap.lambda$null$28(PlayerChunkMap.java:801)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
[13:33:14] [Server thread/WARN]: 	at com.mojang.datafixers.util.Either.mapLeft(Either.java:162)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.PlayerChunkMap.lambda$c$29(PlayerChunkMap.java:784)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
[13:33:14] [Server thread/WARN]: 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkTaskQueueSorter.b(SourceFile:58)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer$a.executeTask(ChunkProviderServer.java:893)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer$a.executeNext(ChunkProviderServer.java:905)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:396)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:325)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.IWorldReader.getChunkAt(IWorldReader.java:87)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:270)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.loadSpawn(MinecraftServer.java:618)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.createWorld(CraftServer.java:1045)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.Bukkit.createWorld(Bukkit.java:536)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.WorldCreator.createWorld(WorldCreator.java:320)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:467)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:448)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.doLoad(WorldManager.java:423)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.utils.WorldManager.loadWorlds(WorldManager.java:728)
[13:33:14] [Server thread/WARN]: 	at com.onarandombox.MultiverseCore.MultiverseCore.onEnable(MultiverseCore.java:310)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:470)
[13:33:14] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:384)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:482)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:298)
[13:33:14] [Server thread/WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885)
[13:33:14] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[13:33:14] [Server thread/WARN]: com.destroystokyo.paper.exception.ServerInternalException: Attempted to place a tile entity (net.minecraft.server.v1_15_R1.TileEntityLectern@619d22b0) at -21,75,358 (Block{minecraft:air}) where there was no entity tile!
```

A similar issue was fixed in the 1.13 update. This occurs when a tile entity is moved/replaced/set without updating the tile entity internally. In 1.13 this was fixed by calling `nmsChunk.d` in setBlock. In 1.15 this has changed to n.m.s.Chunk.removeTileEntity.

I'm not sure if this is the same problem, but it does look awfully familiar.
